landoapi/hgexports.py

```diff
@@ -148,41 +223,136 @@ def commit_description(self) -> bytes:
         finally:
             self.patch.seek(0)
 
-    def write(self, f: io.BytesIO):
-        """Writes whole patch to the specified file object."""
-        try:
-            while 1:
-                buf = self.patch.read(16 * 1024)
-                if not buf:
-                    break
-                f.write(buf)
-        finally:
-            self.patch.seek(0)
-
-    def write_commit_description(self, f: io.BytesIO):
-        """Writes the commit description to the specified file object."""
-        f.write(self.commit_description())
-
-    def write_diff(self, f: io.BytesIO):
-        """Writes the diff to the specified file object."""
+    def get_diff(self) -> bytes:
+        """Return the diff for this patch."""
         try:
+            diff = []
             line_no = 0
             for line in self.patch:
                 line_no += 1
 
                 if self.diff_start_line:
                     if line_no == self.diff_start_line:
-                        f.write(line)
+                        diff.append(line)
                         break
                 else:
                     if self._is_diff_line(line):
-                        f.write(line)
+                        diff.append(line)
                         break
 
             while 1:
                 buf = self.patch.read(16 * 1024)
                 if not buf:
                     break
-                f.write(buf)
+                diff.append(buf)
+
+            return b"".join(diff)
         finally:
             self.patch.seek(0)
+
+    def parse_author_information(self) -> tuple[bytes, bytes]:
+        """Return the author name and email from the patch."""
+        user = self.header("User")
+        if not user:
+            raise ValueError("Patch does not have a `User` header.")
```

Comment:
More accurately stated as
```suggestion
            raise ValueError("Could not determine patch author information from header.")
```

Though if you want you could raise `ValueError` in `get_header`, and keep this same message.